ui/shared/components/panel/family/FamilyReads.jsx

```diff
@@ -197,10 +199,40 @@ const getVariantLocus = (variant, project) => {
   )
 }
 
-const IgvPanel = React.memo((
-  { variant, igvSampleIndividuals, individualsByGuid, project, sampleTypes, rnaReferences, minJunctionEndsVisible },
+const getGeneLocus = (variant, genesById, project) => {
+  if (variant.transcripts) {
+    const geneIds = Object.keys(variant.transcripts)
+    const geneId = geneIds.length > 0 ? geneIds[0] : ''
+    const gene = genesById[geneId]
+    if (gene) {
+      const genomeVersion = (project.genomeVersion === GENOME_VERSION_38) ? 'Grch38' : 'Grch37'
+      const size = gene[`codingRegionSize${genomeVersion}`]
+      return getLocus(
+        gene[`chrom${genomeVersion}`],
+        gene[`start${genomeVersion}`],
+        size ? Math.max(Math.round(size / 3), MIN_LOCUS_RANGE_SIZE) : MIN_LOCUS_RANGE_SIZE,
+        size,
+      )
+    }
+  }
+  // If gene locus is not available, return the variant locus
```

Comment:
if gene locus is not available we should not show the option at all, selecting "gene" and having nothing change is a really bad user experience

Reply:
thinking about it, it might be better to dynamically generate the range dropdown options the way we do for `dnaTrackOptions`. Then you make the value the actual locus itself and pass that - something like 
```
const geneLocus = variant && getGeneLocus(variant, genesById, project)
const rangeOptions = geneLocus ? [{ value: getVariantLocus(variant, project), text: 'Variant' },
  { value: geneLocus, text: 'Gene' },] : null
...
{rangeOptions && ... <RadioGroup ...}
```